docs/proposals/federation.md

```diff
@@ -404,36 +494,107 @@ These are often left implicit by customers, but are worth calling out explicitly
+1. Failure probability: The Decoupled Hierarchical model incorporates
+   numerically more independent pieces of software and configuration
+   than the Monolithic one. But the complexity of each of these
+   decoupled pieces is arguably better contained in the Decoupled
+   model (per standard arguments for modular rather than
+   monolithic software design).  Which of the two models presents
+   higher aggregate complexity and consequent failure probability
+   remains somewhat of an open question.
+2. Scalability: Conceptually the Decoupled Hierarchical model wins here, as
+   each underlying Kubernetes cluster can be scaled completely
+   independently w.r.t. scheduling, node state management, monitoring,
+   network connectivity etc. It is even potentially feasible to stack
+   "Ubernetes" federated clusters (i.e. create federations of
+   federations) should scalability of the independent Federation
+   Control Plane become an issue (although the author does not
+   envision this being a problem worth solving in the short term).
+3. Code complexity: I think that an argument can be made both ways
+   here. It depends on whether you prefer to weave the logic for handling
+   nodes in multiple availability zones and cloud providers within
+   a single logical cluster into the existing Kubernetes control plane
+   code base (which was explicitly not designed for this), or separate
+   it into a decoupled Federation system (with possible code sharing
+   between the two via shared libraries).  The author prefers the
+   latter because it:
+   3.1 Promotes better code modularity and interface design.
+   3.2 Allows (by virtue of the above, but does not mandate) the code
+   bases of Kubernetes and the Federation system to progress largely
+   independently (different sets of developers, different release
+   schedules etc).
+4. Administration complexity: Again, I think that this could be argued
+   both ways.  Superficially it woud seem that administration of a
+   single Monolithic multi-zone cluster might be simpler by virtue of
+   being only "one thing to manage", however in practise each of the
+   underlying availability zones (and possibly cloud providers) has
+   it's own capacity, pricing, hardware platforms, and possibly
+   bureaucratic boudaries (e.g. "our EMEA IT department manages those
+   European clusters").  So explicitly allowing for (but not
+   mandating) completely independent administration of each underlying
+   Kubernetes cluster, and the Federation system itself, in the
+   Decoupled Hierarchical model seems to have real practical benefits
+   that outweigh the superficial simplicity of the Monolithic model.
+5. Application development and deployment complexity:  It's not clear
+   to me that there is any significant difference between the two
+   models in this regard.  Presumably the API exposed by the two
+   different architectures would look very similar, as would the
+   behavior of the deployed applications.
```

Comment:
You omitted one major consideration: Cost of the control plane. For small clusters, multiple full copies of the control plane is a significant resource cost, especially if those are "HA", with multiple instances of each component.